ui/components/multichain/pages/gator-permissions/components/review-gator-permission-item.tsx

```diff
@@ -0,0 +1,601 @@
+
+  /**
+   * The gator permission to display
+   */
+  gatorPermission: StoredGatorPermissionSanitized<
+    Signer,
+    PermissionTypesWithCustom
+  >;
+
+  /**
+   * The function to call when the revoke is clicked
+   */
+  onRevokeClick: () => void;
+};
+
+type PermissionExpandedDetails = Record<
+  string,
+  {
+    translationKey: string;
+    value: string;
+    testId: string;
+  }
+>;
+
+type PermissionDetails = {
+  amountLabel: {
+    translationKey: string;
+    value: string;
+    testId: string;
+  };
+  frequencyLabel: {
+    translationKey: string;
+    valueTranslationKey: string;
+    testId: string;
+  };
+  expandedDetails: PermissionExpandedDetails;
+};
+
+export const ReviewGatorPermissionItem = ({
+  networkName,
+  gatorPermission,
+  onRevokeClick,
+}: ReviewGatorPermissionItemProps) => {
+  const t = useI18nContext();
+  const { permissionResponse, siteOrigin } = gatorPermission;
+  const { chainId } = permissionResponse;
+  const permissionType = permissionResponse.permission.type;
+  const permissionAccount = permissionResponse.address || '0x';
+  const tokenAddress = permissionResponse.permission.data.tokenAddress as
+    | Hex
+    | undefined;
+
+  const [isExpanded, setIsExpanded] = useState(false);
+  const tokensByChain = useSelector(selectERC20TokensByChain);
+  const nativeTokenMetadata = useSelector((state) =>
+    getNativeTokenInfo(state, chainId),
+  ) as TokenMetadata;
+
+  const tokenMetadata: TokenMetadata = useMemo(() => {
+    if (tokenAddress) {
+      const tokenListForChain = tokensByChain?.[chainId]?.data || {};
+      const foundTokenMetadata = getTokenMetadata(
+        tokenAddress,
+        tokenListForChain,
+      );
+      if (foundTokenMetadata) {
+        return {
+          symbol: foundTokenMetadata.symbol || 'Unknown Token',
+          decimals: foundTokenMetadata.decimals || 18,
+          name: foundTokenMetadata.name || 'Unknown Token',
+        };
+      }
+      console.warn(
+        `Token metadata not found for address: ${tokenAddress} for chain: ${chainId}`,
+      );
+      return {
+        symbol: 'Unknown Token',
+        decimals: null,
+        name: 'Unknown Token',
+      };
+    }
+    return {
+      symbol: nativeTokenMetadata.symbol,
+      decimals: nativeTokenMetadata.decimals,
+      name: nativeTokenMetadata.name,
+    };
+  }, [tokensByChain, chainId, tokenAddress, nativeTokenMetadata]);
+
+  /**
+   * Handles the click event for the expand/collapse button
+   */
+  const handleExpandClick = useCallback(() => {
+    setIsExpanded(!isExpanded);
+  }, [isExpanded]);
+
+  /**
+   * Converts a hex value to a decimal value
+   *
+   * @param value - The hex value to convert
+   * @param decimals - The number of decimals to shift the value by
+   * @returns The decimal value
+   */
+  const getDecimalizedHexValue = useCallback(
+    (value: Hex, decimals: number) =>
+      new Numeric(value, 16).toBase(10).shiftedBy(decimals).toString(),
+    [],
+  );
```

Comment:
Looks like this function would benefit from living inside a number utils file and having its own tests. Or perhaps `'@metamask/utils'` also has something e.g. `hexToNumber` that could be useful.